geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/RegionCreateFunction.java

```diff
@@ -94,6 +94,12 @@ public void execute(FunctionContext context) {
     try {
       Region<?, ?> createdRegion = createRegion(cache, regionCreateArgs);
       XmlEntity xmlEntity = new XmlEntity(CacheXml.REGION, "name", createdRegion.getName());
+
+      Region<?, ?> rootRegion = createdRegion.getRootRegion();
```

Comment:
How about just adding a method like getXmlEntityRegionName(Region) and calling that method on line 96 in place of "createdRegion.getName()". This method could do the code that finds the root region and returns the correct name. And that way this code will only create the xmlEntity once.